### ISSUE TITLE
Ad block: block toolbar menu items under 'Change format' have the role 'menuitemcheckbox' instead of 'menuitem'.

### ISSUE THREAD

**Issue by worldomonation** [Automattic/jetpack#32703]
### Impacted plugin

Jetpack

### Quick summary

The 'Change format' popover menu items for the Ad block have the 'menuitemcheckbox' role instead of the expected 'menuitem'. 

WAI-ARIA explains the 'menuitemcheckbox' as supporting multiple states, but as far as I can tell, this functionality is not used in the current implementation of the Ad block.

### Steps to reproduce

1. enable Jetpack Earn if required (Sidebar -> Jetpack -> Settings (on top right) -> Earn -> Toggle on)
2. start a new post.
3. insert the Ad block.
4. select the block if required.
5. on the editor toolbar, click on the "Change format" button.
6. inspect accessibility settings in the toolbar.

![Screenshot 2023-08-28 at 2 54 33 PM](https://github.com/Automattic/jetpack/assets/6549265/383e07e6-f134-45d6-b1a8-37c8d62c7811)

### A clear and concise description of what you expected to happen.

The popover menu items should have the 'menuitem' role, like with every other block we've seen so far.

### What actually happened

The popover menu has the role 'menuitemcheckbox'.

![Screenshot 2023-08-28 at 2 54 04 PM](https://github.com/Automattic/jetpack/assets/6549265/b71d3c77-bc2f-48c7-910e-1fe8fa7fd3f3)

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Atomic

**Comment by worldomonation** [comment]
It shouldn't be escalated, it's a bug without a workaround but the issue is more on the accessibility side and the impact is small.
